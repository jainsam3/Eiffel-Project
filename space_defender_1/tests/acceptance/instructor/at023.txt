--Tests move colliding with projectiles at the end location. 
play(5,20,15,2)
fire
move(D,1)
move(C,6)

play(10,5,10,1)
move(D,1)
fire
move(A,5)
move(D,4)


play(5,12,15,5)
fire
move(A,12)
move(C,12)


play(8,14,15,3)
fire
move(G,7)
move(G,11)
move(D,11)
