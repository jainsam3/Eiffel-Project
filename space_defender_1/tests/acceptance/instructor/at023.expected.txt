  Welcome to Space Defender Version 1.
->play(5,20,15,2)
  state:1.0, ok
      1  2  3  4  5  6  7  8  9 10 11 12 13 14 15 16 17 18 19 20
    A _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _
    B _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _
    C S  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _
    D _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _
    E _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _
->fire
  state:2.0, ok
  The Starfighter fires a projectile at: [C,1]
      1  2  3  4  5  6  7  8  9 10 11 12 13 14 15 16 17 18 19 20
    A _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _
    B _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _
    C S  *  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _
    D _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _
    E _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _
->move(D,1)
  state:3.0, ok
  A projectile moves: [C,2] -> [C,4]
  The Starfighter moves: [C,1] -> [D,1]
      1  2  3  4  5  6  7  8  9 10 11 12 13 14 15 16 17 18 19 20
    A _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _
    B _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _
    C _  _  _  *  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _
    D S  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _
    E _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _
->move(C,6)
  state:4.0, ok
  A projectile moves: [C,4] -> [C,6]
  The Starfighter moves and collides with a projectile: [D,1] -> [C,6]
      1  2  3  4  5  6  7  8  9 10 11 12 13 14 15 16 17 18 19 20
    A _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _
    B _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _
    C _  _  _  _  _  X  _  _  _  _  _  _  _  _  _  _  _  _  _  _
    D _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _
    E _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _
  The game is over. Better luck next time!
->play(10,5,10,1)
  state:5.0, ok
      1  2  3  4  5
    A _  _  _  _  _
    B _  _  _  _  _
    C _  _  _  _  _
    D _  _  _  _  _
    E S  _  _  _  _
    F _  _  _  _  _
    G _  _  _  _  _
    H _  _  _  _  _
    I _  _  _  _  _
    J _  _  _  _  _
->move(D,1)
  state:6.0, ok
  The Starfighter moves: [E,1] -> [D,1]
      1  2  3  4  5
    A _  _  _  _  _
    B _  _  _  _  _
    C _  _  _  _  _
    D S  _  _  _  _
    E _  _  _  _  _
    F _  _  _  _  _
    G _  _  _  _  _
    H _  _  _  _  _
    I _  _  _  _  _
    J _  _  _  _  _
->fire
  state:7.0, ok
  The Starfighter fires a projectile at: [D,1]
      1  2  3  4  5
    A _  _  _  _  _
    B _  _  _  _  _
    C _  _  _  _  _
    D S  *  _  _  _
    E _  _  _  _  _
    F _  _  _  _  _
    G _  _  _  _  _
    H _  _  _  _  _
    I _  _  _  _  _
    J _  _  _  _  _
->move(A,5)
  state:8.0, ok
  A projectile moves: [D,2] -> [D,3]
  The Starfighter moves: [D,1] -> [A,5]
      1  2  3  4  5
    A _  _  _  _  S
    B _  _  _  _  _
    C _  _  _  _  _
    D _  _  *  _  _
    E _  _  _  _  _
    F _  _  _  _  _
    G _  _  _  _  _
    H _  _  _  _  _
    I _  _  _  _  _
    J _  _  _  _  _
->move(D,4)
  state:9.0, ok
  A projectile moves: [D,3] -> [D,4]
  The Starfighter moves and collides with a projectile: [A,5] -> [D,4]
      1  2  3  4  5
    A _  _  _  _  _
    B _  _  _  _  _
    C _  _  _  _  _
    D _  _  _  X  _
    E _  _  _  _  _
    F _  _  _  _  _
    G _  _  _  _  _
    H _  _  _  _  _
    I _  _  _  _  _
    J _  _  _  _  _
  The game is over. Better luck next time!
->play(5,12,15,5)
  state:10.0, ok
      1  2  3  4  5  6  7  8  9 10 11 12
    A _  _  _  _  _  _  _  _  _  _  _  _
    B _  _  _  _  _  _  _  _  _  _  _  _
    C S  _  _  _  _  _  _  _  _  _  _  _
    D _  _  _  _  _  _  _  _  _  _  _  _
    E _  _  _  _  _  _  _  _  _  _  _  _
->fire
  state:11.0, ok
  The Starfighter fires a projectile at: [C,1]
      1  2  3  4  5  6  7  8  9 10 11 12
    A _  _  _  _  _  _  _  _  _  _  _  _
    B _  _  _  _  _  _  _  _  _  _  _  _
    C S  *  _  _  _  _  _  _  _  _  _  _
    D _  _  _  _  _  _  _  _  _  _  _  _
    E _  _  _  _  _  _  _  _  _  _  _  _
->move(A,12)
  state:12.0, ok
  A projectile moves: [C,2] -> [C,7]
  The Starfighter moves: [C,1] -> [A,12]
      1  2  3  4  5  6  7  8  9 10 11 12
    A _  _  _  _  _  _  _  _  _  _  _  S
    B _  _  _  _  _  _  _  _  _  _  _  _
    C _  _  _  _  _  _  *  _  _  _  _  _
    D _  _  _  _  _  _  _  _  _  _  _  _
    E _  _  _  _  _  _  _  _  _  _  _  _
->move(C,12)
  state:13.0, ok
  A projectile moves: [C,7] -> [C,12]
  The Starfighter moves and collides with a projectile: [A,12] -> [C,12]
      1  2  3  4  5  6  7  8  9 10 11 12
    A _  _  _  _  _  _  _  _  _  _  _  _
    B _  _  _  _  _  _  _  _  _  _  _  _
    C _  _  _  _  _  _  _  _  _  _  _  X
    D _  _  _  _  _  _  _  _  _  _  _  _
    E _  _  _  _  _  _  _  _  _  _  _  _
  The game is over. Better luck next time!
->play(8,14,15,3)
  state:14.0, ok
      1  2  3  4  5  6  7  8  9 10 11 12 13 14
    A _  _  _  _  _  _  _  _  _  _  _  _  _  _
    B _  _  _  _  _  _  _  _  _  _  _  _  _  _
    C _  _  _  _  _  _  _  _  _  _  _  _  _  _
    D S  _  _  _  _  _  _  _  _  _  _  _  _  _
    E _  _  _  _  _  _  _  _  _  _  _  _  _  _
    F _  _  _  _  _  _  _  _  _  _  _  _  _  _
    G _  _  _  _  _  _  _  _  _  _  _  _  _  _
    H _  _  _  _  _  _  _  _  _  _  _  _  _  _
->fire
  state:15.0, ok
  The Starfighter fires a projectile at: [D,1]
      1  2  3  4  5  6  7  8  9 10 11 12 13 14
    A _  _  _  _  _  _  _  _  _  _  _  _  _  _
    B _  _  _  _  _  _  _  _  _  _  _  _  _  _
    C _  _  _  _  _  _  _  _  _  _  _  _  _  _
    D S  *  _  _  _  _  _  _  _  _  _  _  _  _
    E _  _  _  _  _  _  _  _  _  _  _  _  _  _
    F _  _  _  _  _  _  _  _  _  _  _  _  _  _
    G _  _  _  _  _  _  _  _  _  _  _  _  _  _
    H _  _  _  _  _  _  _  _  _  _  _  _  _  _
->move(G,7)
  state:16.0, ok
  A projectile moves: [D,2] -> [D,5]
  The Starfighter moves: [D,1] -> [G,7]
      1  2  3  4  5  6  7  8  9 10 11 12 13 14
    A _  _  _  _  _  _  _  _  _  _  _  _  _  _
    B _  _  _  _  _  _  _  _  _  _  _  _  _  _
    C _  _  _  _  _  _  _  _  _  _  _  _  _  _
    D _  _  _  _  *  _  _  _  _  _  _  _  _  _
    E _  _  _  _  _  _  _  _  _  _  _  _  _  _
    F _  _  _  _  _  _  _  _  _  _  _  _  _  _
    G _  _  _  _  _  _  S  _  _  _  _  _  _  _
    H _  _  _  _  _  _  _  _  _  _  _  _  _  _
->move(G,11)
  state:17.0, ok
  A projectile moves: [D,5] -> [D,8]
  The Starfighter moves: [G,7] -> [G,11]
      1  2  3  4  5  6  7  8  9 10 11 12 13 14
    A _  _  _  _  _  _  _  _  _  _  _  _  _  _
    B _  _  _  _  _  _  _  _  _  _  _  _  _  _
    C _  _  _  _  _  _  _  _  _  _  _  _  _  _
    D _  _  _  _  _  _  _  *  _  _  _  _  _  _
    E _  _  _  _  _  _  _  _  _  _  _  _  _  _
    F _  _  _  _  _  _  _  _  _  _  _  _  _  _
    G _  _  _  _  _  _  _  _  _  _  S  _  _  _
    H _  _  _  _  _  _  _  _  _  _  _  _  _  _
->move(D,11)
  state:18.0, ok
  A projectile moves: [D,8] -> [D,11]
  The Starfighter moves and collides with a projectile: [G,11] -> [D,11]
      1  2  3  4  5  6  7  8  9 10 11 12 13 14
    A _  _  _  _  _  _  _  _  _  _  _  _  _  _
    B _  _  _  _  _  _  _  _  _  _  _  _  _  _
    C _  _  _  _  _  _  _  _  _  _  _  _  _  _
    D _  _  _  _  _  _  _  _  _  _  X  _  _  _
    E _  _  _  _  _  _  _  _  _  _  _  _  _  _
    F _  _  _  _  _  _  _  _  _  _  _  _  _  _
    G _  _  _  _  _  _  _  _  _  _  _  _  _  _
    H _  _  _  _  _  _  _  _  _  _  _  _  _  _
  The game is over. Better luck next time!
